pkg/scheduler/framework/plugins/interpodaffinity/scoring.go

```diff
@@ -58,8 +58,8 @@ func (m scoreMap) processTerm(term *framework.AffinityTerm, weight int32, pod *v
 }
 
 func (m scoreMap) processTerms(terms []framework.WeightedAffinityTerm, pod *v1.Pod, nsLabels labels.Set, node *v1.Node, multiplier int32, enableNamespaceSelector bool) {
-	for _, term := range terms {
-		m.processTerm(&term.AffinityTerm, term.Weight, pod, nsLabels, node, multiplier, enableNamespaceSelector)
+	for i := range terms {
+		m.processTerm(&terms[i].AffinityTerm, terms[i].Weight, pod, nsLabels, node, multiplier, enableNamespaceSelector)
```

Comment:
iiuc, the issue could only happen when you manage a pointer and `append` that pointer, I checked the code and cannot see any problem without this change, if you believe this is an issue could you pls wrote a testcase to guard this?

Reply:
Yes as you said from code perspective there is no such issue as the same address is being used for each new value or `term` but we are using the reference to send it's `AffinityTerm` to the `processTerm` this will still work 

but incase in future we extend the usage and accidentally use this way of referencing for `term` where the value stored at that address is being used further then this will create an issue as it will hold the last assigned value (if declared outside the scope of the loop), also in case of append to the pointer problems could arise so as a best practice we can remove such usage os loop variables by referencing (address)

In all other components under this repository, i have seen this as a standard practice as well.

A test case might not able to catch any error in the code in its present state @chendave 

As it's not a bug currently should i change the kind of the PR to `enhancement` or `refactor` ?

Reply:
I am fine with this change, leaving this to @alculquicondor as I don't have privilege to approve a PR.
> As it's not a bug currently should i change the kind of the PR to enhancement or refactor ?

bug is okay, as you want to avoid a potential issue.

Reply:
This is really a non-issue. It's increasing the line length with no clear benefit.

Reply:
I agree with @alculquicondor. There is no correctness issue here as `processTerm()` is immediately processed and returned.

Using the looped variable is problematic when you store them in a slice and return later for further usage - it'd end up with identical objects stored multiple times. But it's not the case here.